### PR TITLE
Hypercubes as array

### PR DESCRIPTION
**Pull request by ghislainfourny**

### CODE REVIEW

File: queries/modules/io/28/apps/reports.jq [hunk 1]
```diff
@@ -75,26 +77,31 @@ declare function reports:concepts-checks($report as object)
 as object*
 {
     (: check whether all indexes match the concept name -> index : { Name: index } :)
```
Comment (dknochen) [inline queries/modules/io/28/apps/reports.jq]: $all-hypercube-mappings is an array now. Maybe, we should remove this superfluous check?

File: queries/modules/io/28/modules/xbrl/hypercubes.jq [hunk 2]
```diff
@@ -61,9 +61,29 @@ declare variable $hypercubes:IMPLIED_TABLE as string := "xbrl28:ImpliedTable";
  :
  : @return all hypercubes.
  :)
+declare function hypercubes:hypercube-names-for-components($components as object*) as string*
+{
```
Comment (dknochen) [inline queries/modules/io/28/modules/xbrl/hypercubes.jq]: you could have leveraged the hypercubes:hypercubes-for-components() function here ;).

File: queries/modules/io/28/modules/xbrl/hypercubes.jq [hunk 3]
```diff
@@ -80,7 +100,11 @@ declare function hypercubes:hypercubes-for-components(
     $names as string*
 ) as object*
 {
```
Comment (dknochen) [inline queries/modules/io/28/modules/xbrl/hypercubes.jq]: same here actually.

File: queries/private/migration/db10.jq [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import module namespace export = "http://apps.28.io/reports-export";
+
+for $report in collection("reports")
```
Comment (dknochen) [inline queries/private/migration/db10.jq]: could we add this script to a seperate pull request? I think, the cell store manager does not work with arrays, yet. Also, we need to change the report templates in that separate PR.

**Comment by dknochen** [comment]
@ghislainfourny very good :) see some comments above. We can discuss them offline if you want. thank you :)

File: queries/modules/io/28/apps/reports.jq (hunk 1 above)
Comment (ghislainfourny) [inline queries/modules/io/28/apps/reports.jq]: DONE

File: queries/modules/io/28/modules/xbrl/hypercubes.jq (hunk 2 above)
Comment (ghislainfourny) [inline queries/modules/io/28/modules/xbrl/hypercubes.jq]: DONE

File: queries/modules/io/28/modules/xbrl/hypercubes.jq (hunk 3 above)
Comment (ghislainfourny) [inline queries/modules/io/28/modules/xbrl/hypercubes.jq]: DONE

File: queries/private/migration/db10.jq (hunk 4 above)
Comment (ghislainfourny) [inline queries/private/migration/db10.jq]: DONE

File: tasks/config.js [hunk 5]
```diff
@@ -127,7 +127,8 @@ var config =
                 'queries/private/UpdateReportSchema.jq',
                 'queries/private/cleanupTestUserReports.jq'
             ],
-            'queries/private/initElasticSearch.jq'
+            'queries/private/initElasticSearch.jq',
+            'queries/private/migration/db10.jq'
```
Comment (dknochen) [inline tasks/config.js]: this needs to go to the other PR. Not sure, why it doesn't fail.

**Comment by dknochen** [comment]
@ghislainfourny one last comment. Then, its good to be merged :)

**Pull request merged**